elasticai/creator/nn/quantized_grads/fixed_point/_quantize_to_fixed_point.py

```diff
@@ -0,0 +1,27 @@
+import torch
+
+from elasticai.creator.nn.quantized_grads.fixed_point import FixedPointConfigV2
+
+
+def quantize(number: torch.Tensor, fxp_conf: FixedPointConfigV2) -> torch.Tensor:
+    return _round(clamp(number, fxp_conf), fxp_conf)
+
+
+def _round(number: torch.Tensor, fxp_conf: FixedPointConfigV2) -> torch.Tensor:
+    if fxp_conf.stochastic_rounding:
```

Comment:
That if statement will not be found during tracing, therefore we might prefer to use two different functions. Following this thought, we'd end up with two different models, one using fxp rounding while the other doesn't. Not sure if that'd be ok for you, but i think it's the cleaner approach. That way you'd not even have to add the stochastic rounding flag to fxp_config and the fxp_config could just be `(total_Bits, frac_bits)`.

Also, from experience flag arguments like this often make it hard to know which rounding method is actually used, especially if the `fxp_conf` object is mutable.

For now i think it'd even be fine to just add a comment, that you'd have to be aware of this when tracing. Or we flag entire modules as not being fit for tracing. Maybe @mokouMonday has an opinion here (i guess you're one of the people for which that matters).

Rest looks good ;).

Reply:
I made the FixedPointConfigV2 immutable. I changed the functions, so tracing will allow to see if deterministic or stochastic rounding is used.

Reply:
Making it immutable is certainly an improvement here, but i have the impression that you might misunderstand how tracing works. What i wanted to say above is that tracing will not see that there is an if statement in python code. I think immutability solves the issue of the traced version not behaving like the code it was traced from though.